src/harness/unittests/tscWatchMode.ts

```diff
@@ -120,24 +120,50 @@ namespace ts.tscWatch {
         }
 
         function getOutputAtFailedMessage(caption: string, expectedOutput: string) {
-            return `Expected ${caption}: ${expectedOutput} at ${index} in ${JSON.stringify(outputs)}`;
+            return `Expected ${caption}: ${JSON.stringify(expectedOutput)} at ${index} in ${JSON.stringify(outputs)}`;
         }
 
-        function getWatchDiagnosticWithoutDate(diagnosticMessage: DiagnosticMessage) {
-            return ` - ${flattenDiagnosticMessageText(getLocaleSpecificMessage(diagnosticMessage), host.newLine)}${host.newLine + host.newLine + host.newLine}`;
+        function getWatchDiagnosticWithoutDate(diagnostic: Diagnostic) {
+            return ` - ${flattenDiagnosticMessageText(diagnostic.messageText, host.newLine)}${host.newLine + host.newLine + host.newLine}`;
         }
     }
 
     function checkOutputErrorsInitial(host: WatchedSystem, errors: ReadonlyArray<Diagnostic>, disableConsoleClears?: boolean, logsBeforeErrors?: string[]) {
-        checkOutputErrors(host, /*logsBeforeWatchDiagnostic*/ undefined, Diagnostics.Starting_compilation_in_watch_mode, logsBeforeErrors, errors, disableConsoleClears, Diagnostics.Compilation_complete_Watching_for_file_changes);
+        checkOutputErrors(
+            host,
+            /*logsBeforeWatchDiagnostic*/ undefined,
+            createCompilerDiagnostic(Diagnostics.Starting_compilation_in_watch_mode),
+            logsBeforeErrors,
+            errors,
+            disableConsoleClears,
+            errors.length === 1
+                ? createCompilerDiagnostic(Diagnostics.Found_1_error)

```

Comment:
Instead of doing this here, can you add the move this to https://github.com/Microsoft/TypeScript/pull/22254/files#diff-09ed66c9ac499e7fd991780ed486d800R96 and check ```if (postErrorsWatchDiagnostics.length) ``` and do this there so that its in one place.